### ISSUE TITLE
SonarCSS fails on parsing a .cshtml file.

### ISSUE THREAD

**Issue by Vossekop** [SonarSource/sonar-css#197]
The following information is displayed in my build log:

INFO: ------------- Run sensors on module UserInterface
INFO: 2/2 source files have been analyzed
INFO: Sensor SonarCSS Metrics [cssfamily]
INFO: Sensor SonarCSS Metrics [cssfamily] (done) | time=109ms
INFO: Sensor SonarCSS Rules [cssfamily]
##[error]ERROR: Failed to parse file:///D:/a/1/s/UserInterface/Views/Account/_UserAccount.cshtml, line 69, At-rule without name
ERROR: Failed to parse file:///D:/a/1/s/UserInterface/Views/Account/_UserAccount.cshtml, line 69, At-rule without name
INFO: Sensor SonarCSS Rules [cssfamily] (done) | time=12108ms

Some more information regarding my environment:

Running SonarCloud tasks (v1.6.3) on Azure DevOps Services. I would prefer it does not fail on parsing but I will accept a work around where I can ignore this file type only for SonarCSS.

Kind regards,

Jeroen

**Comment by vilchik-elena** [comment]
Excluding this file from analysis should remove parsing error.

Can you share the code where parsing fails?

**Comment by Vossekop** [comment]
Excluding it from analysis completely is not a very good solution in my opinion, I still want other sensors that don't break to analyze it. I would preferably have the ability to exclude it only for the SonarCSS sensor, but this does not seem possible. Strange thing is that this filetype is not even included in the SonarCSS settings anyway, and still it tries to scan it.

As for the contents, I will ask the relevant team if I can safely share the code here, or at least line 69.

**Comment by vilchik-elena** [comment]
Indeed this file is not included in CSS settings, but as it's included in HTML settings we try to analyse it.

**Comment by Vossekop** [comment]
The offending line is as follows:

```
<small id="temporary_password_info" style="@(!Model.IsPasswordResetAllowed ? "display: none;" : string.Empty)">@Resources.User_TemporaryPassword</small>
```

**Comment by vilchik-elena** [comment]
We might consider swallow parsing errors when they occur on non-css files. Just logging them in debug might be good enough

**Comment by Vossekop** [comment]
I think that would be preferable to breaking yes, and it is also better than having to exclude the entire file.

**Comment by Rychu-Pawel** [comment]
We are affected by this issue as well. Do you guys have any ETA for switching logging to debug for this?

**Comment by ashu16815** [comment]
HI Team,

WE are having same issue. 

+1

Ankit

**Comment by vilchik-elena** [comment]
Am I right, that this is issue is critical to you because build on Azure fails when there is ERROR log?

**Comment by ashu16815** [comment]
Yes , You are right.

**Comment by Rychu-Pawel** [comment]
For us it is a bit different. The step throws errors but somehow the whole build is successfull:

![obraz](https://user-images.githubusercontent.com/22656474/66327066-77d72f00-e92a-11e9-87a5-0daf393d7e8e.png)

But it is annoying as build result is highlithed in red.

**Comment by Bouke** [comment]
Same here. The output from this scanner is logged as "error", while it doesn't fail the build. It shouldn't fail the build of course, so that's good. However the "errors" are unneeded distractions as they don't offer any insights. Other than that cshtml cannot be parsed as css, which is hardly surprising.

**Comment by chicadilly** [comment]
When we upgraded our SonarQube server (and thereby updating our SonarCSS plugin), we started seeing this error come up:

ERROR: Failed to parse file:///C:/agent/_work/9/s/product/channels/ConsumerBanking/UI/Areas/Ach/Views/UploadBatch/Edit.cshtml, line 140, Unclosed bracket

It is flagged as ERROR in the build log, but it does not fail the build.  

We would like to see this error flow back to the SonarQube server so it trips the Quality Gate.  Is there something I can tweak to get errors such as this to flow back to the server from the SonarCSS plugin?
